geomstats/learning/ahc.py

```diff
@@ -0,0 +1,118 @@
+"""The Agglomerative Hierarchical Clustering (AHC) on manifolds."""
+
+from sklearn.cluster import AgglomerativeClustering
+
+import geomstats.backend as gs
+
+
+class AgglomerativeHierarchicalClustering(AgglomerativeClustering):
+    """The Agglomerative Hierarchical Clustering on manifolds.
+
+    Recursively merges the pair of clusters that minimally increases
+    a given linkage distance.
+
+    Parameters
+    ----------
+    n_clusters : int or None, default=2
+        The number of clusters to find. It must be ``None`` if
+        ``distance_threshold`` is not ``None``.
+    distance : str or callable, default='euclidean'
```

Comment:
Cool! Could you add an exemple section in the docstring so we know how to pass a geomstats geodesic distance? As in https://github.com/geomstats/geomstats/blob/6cf2aba8f917718028e445207eb3705fb56c98ab/geomstats/learning/online_kmeans.py#L142 for example:
```
    Example
    -------
    >>> from geomstats.geometry.hypersphere import Hypersphere
    >>> from geomstats.learning.ahc import AgglomerativeHierarchicalClustering
    >>> sphere = Hypersphere(dimension=2)
    >>> X = sphere.random_uniform(10)
    >>> distance = sphere.metric.dist
    >>> AgglomerativeHierarchicalClustering(distance=distance)
    >>> clustering.fit(X)
```